### PR TITLE
include: storage: flash_map: _NODE macros

### PR DESCRIPTION
**Pull request by michalek-no**
adds size, offset and divice access macros that are operating with nodes as an argumet.

### CODE REVIEW

**Review by de-nordic — COMMENTED** [review]
LGTM.
Nevertheless the Flash Map API is considered stable API (https://docs.zephyrproject.org/2.7.5/reference/api/overview.html#api-overview) and therefore tests scenarios are required by the change.

**Review by sigvartmh — APPROVED** [review]

**Comment by michalek-no** [comment]
rebase + tests

**Review by de-nordic — DISMISSED** [review]

**Review by carlescufi — DISMISSED** [review]

File: include/zephyr/storage/flash_map.h [hunk 1]
```diff
@@ -310,6 +319,15 @@ uint8_t flash_area_erased_val(const struct flash_area *fa);
  */
 #define FIXED_PARTITION_SIZE(label) DT_REG_SIZE(DT_NODELABEL(label))
 
+/**
+ * Get fixed-partition size for DTS node
+ *
+ * @param node DTS node of a partition
+ *
+ * @return fixed-partition offset, as defined for the partition in DTS.
```
Comment (kartben) [inline include/zephyr/storage/flash_map.h:327]:
```suggestion
 * @return fixed-partition size, as defined for the partition in DTS.
```

File: tests/subsys/storage/flash_map/src/main.c [hunk 2]
```diff
@@ -190,4 +191,15 @@ ZTEST(flash_map, test_flash_area_erased_val)
 	flash_area_close(fa);
 }
 
+ZTEST(flash_map, test_fixed_partition_noded_macros)
```
Comment (kartben) [inline tests/subsys/storage/flash_map/src/main.c:194]:
```suggestion
ZTEST(flash_map, test_fixed_partition_node_macros)
```

**Review by kartben — CHANGES_REQUESTED** [review]
couple typos spotted

**Review by carlescufi — APPROVED** [review]

**Review by kartben — APPROVED** [review]

**Pull request merged**